wallet/src/de/schildbach/wallet/ui/dashpay/utils/ProfilePictureDisplay.kt

```diff
@@ -1,135 +1,43 @@
-                        .transform(ProfilePictureTransformation.create(avatarUrlStr))
-                        .placeholder(defaultAvatar)
-                        .apply {
-                            takeIf { !disableTransition }
-                                    ?.apply(transition(withCrossFade()))
-                        }.into(avatarView)
-            } else {
-                displayDefault(avatarView, username, fontSize, listener)
-            }
-        }
-
-        @JvmStatic
-        fun displayDefault(avatarView: ImageView, username: String, listener: OnResourceReadyListener? = null) {
-            displayDefault(avatarView, username, calcFontSize(avatarView), listener)
-        }
-
-        private fun displayDefault(avatarView: ImageView, username: String, fontSize: Int, listener: OnResourceReadyListener?) {
-            val defaultAvatar: Drawable? = getDrawable(avatarView.context, username[0], fontSize)
-            avatarView.setImageDrawable(defaultAvatar)
-            listener?.onResourceReady(defaultAvatar)
-        }
-
-        const val FONT_SIZE_RATIO: Float = 30f / 64f
-
-        private fun calcFontSize(avatarView: ImageView): Int {
-            return (avatarView.layoutParams.width * FONT_SIZE_RATIO).toInt()
-        }
-
-        fun removePicZoomParameter(url: String): Uri {
-            return removeParameter(Uri.parse(url), "dashpay-profile-pic-zoom")
-        }
-
-        @Suppress("SameParameterValue")
-        private fun removeParameter(uri: Uri, key: String): Uri {
-            val newUriBuilder = uri.buildUpon().clearQuery()
-            for (param in uri.queryParameterNames) {
-                newUriBuilder.appendQueryParameter(param,
-                        if (param == key) {
-                            continue
-                        } else {
-                            uri.getQueryParameter(param)
-                        })
-            }
-            return newUriBuilder.build()
-        }
-    }
-
-    interface OnResourceReadyListener {
-        fun onResourceReady(resource: Drawable?)
+import org.dash.wallet.common.ui.avatar.ProfilePictureDisplay
+
+fun ProfilePictureDisplay.Companion.display(avatarView: ImageView, dashPayProfile: DashPayProfile?, hideIfProfileNull: Boolean = false) {
+    display(avatarView, dashPayProfile, hideIfProfileNull, false, null)
+}
+
+fun ProfilePictureDisplay.Companion.display(avatarView: ImageView, dashPayProfile: DashPayProfile?, hideIfProfileNull: Boolean = false, disableTransition: Boolean, listener: ProfilePictureDisplay.OnResourceReadyListener?) {
+    if (dashPayProfile != null) {
+        avatarView.visibility = View.VISIBLE
+        display(
+            avatarView,
+            dashPayProfile.avatarUrl,
+            dashPayProfile.avatarHash,
+            dashPayProfile.username,
+            disableTransition,
+            listener
+        )
+    } else if (hideIfProfileNull) {
+        avatarView.visibility = View.GONE
```

Comment:
Most of `ProfilePictureDisplay` is moved to common and only parts that depend on `DashPayProfile` and `dpp` are left in the wallet as extensions.